src/dnsimple/certificates.rs

```diff
@@ -102,13 +108,17 @@ pub struct LetsEncryptPurchasePayload {
     pub name: String,
     /// The certificate alternate names (i.e. ["docs.example.com", "status.example.com"])
     pub alternate_names: Vec<String>,
+    /// Signature algorithm to be used.
+    pub signature_algorithm: LetsEncryptSignatureAlgorithm,
```

Comment:
@DXTimer Curious to hear what you think about using `Option` and/or `Default` for serialising, deserialising, and initialising these structs.

Reply:
I think it makes a lot of sense to use `Option` and it would make for a better DX.

Reply:
We would need to add tests to ensure that nothing is passed in the request body if no value is provided.

Reply:
As Rust requires all fields to be set and we use fields directly instead of e.g. builder pattern, I think of two main options:

- We can impl Default for LetsEncryptSignatureAlgorithm so that users can use `Default::default` for that field.
- Use Option so that users can use `None`. Upon serialising, we leave field blank if `None`, letting server decide the default.

Using Option and `None` might be more confusing as the API states that omitting simply uses the default, and is not a third variant (which `None` might imply). A downside is that any Default impl must be in sync with the API, so perhaps that's a reason to use `Option`. We don't appear to deserialise this struct currently, so we don't need to consider what missing and `null` means right now.

Another option is to derive Default for the struct so all optional fields can be filled automatically using `..Default::default()`, which is not necessary in this case with only one optional field but something to consider generally.

Reply:
I am more in favor of option two as it doesn't have the tight coupling the first option has. That being said, the decision is up to you.

Reply:
Done 👍